### DISCUSSION TITLE
Tailwind does not work in forms for any Livewire components

### DISCUSSION THREAD

**Discussion by AutobookNft** [filamentphp/filament#11973] [Help]
### Package

Form builder

### Package Version

3.2.42

### How can we help you?

Hello everyone, and thanks for any help you can give me

Within Livewire components, which extend HasForm, Tailwind classes are ignored.
This is pretty much identical to the documentation

`<?php

namespace App\Livewire;

use App\Models\Post;
use Filament\Forms\Components\RichEditor;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Forms\Form;
use Illuminate\Contracts\View\View;
use Livewire\Component;

class CreatePost extends Component implements HasForms
{
    use InteractsWithForms;

    public ?array $data = [];

    public function mount(): void
    {
        $this->form->fill();
    }

    public function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('title')
                    ->required(),
                RichEditor::make('content')
                    ->required()
                    ->toolbarButtons([
                        'attachFiles',
                        'blockquote',
                        'bold',
                        'bulletList',
                        'codeBlock',
                        'h2',
                        'h3',
                        'italic',
                        'link',
                        'orderedList',
                        'redo',
                        'strike',
                        'underline',
                        'undo',
                    ]),
            ])
            ->statePath('data')
            ->model(Post::class);
    }

    public function create(): void
    {
        dd($this->form->getState());
    }

    public function render(): View
    {
        return view('livewire.create-post');
    }
}`

the classes that don't work are the ones I add to the div, not the ones internal to Filament
these classes are ignored                 inline css with style works
---------------xxxxxxxxxx--------- vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
```
<div class = "p-4 mt-72" style ="margin-top:4rem; padding:8rem;">
    <form wire:submit="create">
        {{ $this->form }}

        <button type="submit" class ="bg-lime-400">
            Submit
        </button>

    </form>

    <x-filament-actions::modals />
</div>
```

this is the form with the css style
_(Furthermore, the richttext, inside the forms included in this way, is "closed")_
![form](https://github.com/filamentphp/filament/assets/100548780/e8958727-8a3d-49c0-b69a-c21db88710a0)

this is the form with the Tailwind css
![without css style form](https://github.com/filamentphp/filament/assets/100548780/ea49cb24-d049-421c-8344-c323d4fd0825)

this is my tailwind.config.js file
```
const defaultTheme = require("tailwindcss/defaultTheme");

/** @type {import('tailwindcss').Config} */

module.exports = {
    content: [
        "./vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php",
        "./vendor/laravel/jetstream/**/*.blade.php",
        "./storage/framework/views/*.php",
        "./resources/views/**/*.blade.php",
        "./resources/**/*.vue",
        './vendor/filament/**/*.blade.php',
        "./node_modules/tw-elements/dist/js/**/*.js",
        './resources/views/filament/**/*.blade.php',
    ],

    theme: {
        extend: {
            fontFamily: {
                sans: ["Nunito", ...defaultTheme.fontFamily.sans],
                'roboto': ['Roboto', 'sans-serif'],
                'inter': ['Inter', 'sans-serif'],

            },
            fontSize: {
                xxs: "0.5rem",
            },
            colors: {
                c123de: "#c123de",
                a20dbd: "#a20dbd",
                a511c0: "#a511c0",
            },
        },

        container: {
            center: true,
            padding: "1rem",
            screens: {
                sm: "100%",
                md: "100%",
                lg: "800px",
                xl: "800px",
                "2xl": "1000px",
                "3xl": "1200px",
                "4xl": "1400px",
                "5xl": "1600px",
            },
        },

        darkMode: "class",

        screens: {
            xxs: "480px",
            // => @media (min-width: 480px) { ... }

            xs: "540px",
            // => @media (min-width: 540px) { ... }

            sm: "640px",
            // => @media (min-width: 640px) { ... }

            md: "768px",
            // => @media (min-width: 768px) { ... }

            ll: "850px",
            // => @media (min-width: 868px) { ... }

            lg: "1024px",
            // => @media (min-width: 1024px) { ... }

            xl: "1280px",
            // => @media (min-width: 1280px) { ... }

            xl1: "1400px",
            // => @media (min-width: 1400px) { ... }

            "2xl": "1536px",
            // => @media (min-width: 1536px) { ... }

            "3xl": "1700px",
            // => @media (min-width: 1700px) { ... }

            "4xl": "1800px",
            // => @media (min-width: 1800px) { ... }
        },
    },

    plugins: [
        require("tw-elements/dist/plugin.cjs"),
        require("@tailwindcss/forms"),
        require("@tailwindcss/typography"),
    ],
};
```

**Comment by CrisCrassus** [comment]
Hi there, did you ever find a solution for this?  I'm wracking my brains for multiple projects and cannot find a solution.

**Comment by lordjack** [comment]
It worked for me. You need to install Theme Custom as it shows the  [link](https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme), and then you call CSS on component as it shows the example below

You need to make sure tailwind is configured properly (included the filament styles, included app.css etc).

```
@vite('resources/css/app.css') 
<x-filament-panels::page> 
    <div class="grid grid-cols-3 gap-3"> 
        <div class="col-span-2 bg-white rounded border p-5">sd</div> 
        <div class="col-span-1 bg-white rounded border p-5">sd</div> 
    </div> 
</x-filament-panels::page> 
```

**Reply by superuser-me** [reply]
Thank you,
In my livewire component which i mounted inside infolist,
dark mode tailwind styles were not applied similar to the above question,
i have added this like you have said and it has fixed the issue,
spend 3 hours debugging this issue.
@lordjack You're a real lord! lol , :)
`@vite('resources/css/app.css') 
`
